Remove redundant includes. Conform to header include guidelines.

practicalswift opened the pull request:
From the header include guidelines ([developer-notes.md](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#source-code-organization)):

> "One exception is that a `.cpp` file does not need to re-include the includes already included in its corresponding `.h` file."

Covered in this PR:
* `rpc/util.h` includes `pubkey.h` + `utilstrencodings.h`. `rpc/util.cpp` includes `rpc/util.h`.
* `util.h` includes `fs.h`. `util.cpp` includes `util.h`.

sipa commented on src/rpc/util.cpp:
```diff
@@ -4,11 +4,9 @@
 #include <rpc/protocol.h>
 #include <rpc/util.h>
 #include <tinyformat.h>
-#include <utilstrencodings.h>
```
`utilstrencodings.h` seems to be included in `rpc/util.h` for no reason. Maybe it's better to remove it there?

practicalswift committed:
Remove redundant includes. Conform to header include guidelines.

From the header include guidelines (developer-notes.md):

"One exception is that a `.cpp` file does not need to re-include the
 includes already included in its corresponding `.h` file."

* rpc/util.h includes pubkey.h + utilstrencodings.h. rpc/util.cpp includes rpc/util.h.
* util.h includes fs.h. util.cpp includes util.h.

practicalswift commented:
@sipa Thanks for reviewing! Feedback addressed. Please re-review :-)

Empact commented:
utACK 7ef46d0

eklitzke reviewed (approved):
utACK 7ef46d063aaeb3411047ef249adcfeab63a2d6e9

Curious if you've tried using `iwyu` at all. There are so many header/forward include violations in the source code that it's hard to imagine the output would be comprehensible, but it can help with this kind of task.

practicalswift commented:
@eklitzke No I haven't tried `iwyu` yet. These violations were found using my `lint-includes.sh` script which has been submitted as PR #11878 :-)

ryanofsky commented:
> When this PR is merged I can try iwyu and see if it finds something worth fixing :-)

I sometimes use IWYU. I posted steps to run it with bitcoin at https://github.com/bitcoin/bitcoin/pull/11878#issuecomment-351610027

sipa commented:
utACK 7ef46d063aaeb3411047ef249adcfeab63a2d6e9
